apps/test/unit/templates/instructions/InstructionsWithWorkspaceTest.jsx

```diff
@@ -135,4 +155,249 @@ describe('InstructionsWithWorkspace', () => {
       expect(setInstructionsMaxHeightAvailable).not.to.have.been.called;
     });
   });
+
+  // This is a collection of tests that deep-render InstructionsWithWorkspace and its children,
+  // to check interactions between them.
+  describe('integration tests', () => {
+    beforeEach(() => {
+      stubRedux();
+      // Setup minimum redux config for these integration tests
+      registerReducers({
+        instructions: instructionsReducer,
+        isRtl: isRtlReducer,
+        pageConstants: pageConstantsReducer
+      });
+    });
+
+    afterEach(() => {
+      restoreRedux();
+    });
+
+    // This set of integration tests checks that we don't render the instructions when they
+    // are not available or not needed.
+    // These were originally written as integration tests to cover moving this logic from
+    // TopInstructions up to InstructionsWithWorkspace, but could be changed to unit tests
+    // later on.
+    describe('hiding the instructions', () => {
```

Comment:
Here are the integration tests that check whether the instructions and resize bar are hidden in the correct circumstances.